src/service/ampdoc-impl.js

```diff
@@ -418,20 +418,47 @@ export class AmpDoc {
     return v == null ? null : v;
   }
 
+  /**
+   * Returns map of an ampdoc's meta name value to content value
+   * @return {!Object<string, string>}
+   */
+  getMeta() {
+    const metaMap = map();
+    if (!this.win.document || !this.win.document.head) {
+      return metaMap;
+    }
+
+    const metas = this.win.document.head.querySelectorAll('meta[name]');
+    iterateCursor(metas, meta => {
+      const content = meta.getAttribute('content');
+      const name = meta.getAttribute('name');
+      if (!name || content === null) {
+        return;
+      }
+
+      // Retain only the first meta content value for a given name, consistent
+      // with getMetaByName.
```

Comment:
Not sure this is safe to do since there may be code that's relying on the existing "concat" behavior (instead of only retaining the first meta content).

Reply:
I think it is safe, actually. There are very few uses of DocInfo.metaTags, and the array output is never used (in fact, in one place, the array behavior isn't even considered and I was surprised type checking didn't report an error).

Reply:
Thanks for checking.